### PR TITLE
Implement error checks

### LINKED ISSUE #7: Error checking
We cannot raise Python exceptions from our custom XLA calls, so we need some other mechanism to tell users when an MPI exception occurred.

I see a couple of different ways forward:

1. Do a hard abort with a message to `stderr` whenever an MPI function returns an error code (probably best combined with a call to `MPI_Abort`)
2. Return the error code and let the user code handle it

Not sure how useful 2 is though, since you can't raise from jitted code.

### PR DESCRIPTION
**Pull request by dionhaefner**
Fixes #7

### CODE REVIEW

File: mpi4jax/cython/mpi_xla_bridge.pyx [hunk 1]
```diff
@@ -24,6 +26,26 @@ from mpi4py.libmpi cimport (
 MPI_STATUS_IGNORE_ADDR = int(<uint64_t>MPI_STATUS_IGNORE)
 
 
+#
+# Error handling
+#
+
+cdef inline int abort_on_error(int ierr, MPI_Comm comm, unicode mpi_op) nogil:
+    cdef int rank
```
Comment (PhilipVinc) [inline mpi4jax/cython/mpi_xla_bridge.pyx:34]: Can you just put this cdef below the `    if ierr == MPI_SUCCESS:` line? 
Makes it clearer for me.

**Comment by PhilipVinc** [comment]
You want to add something else?
Otherwise this looks good to be merged already

File: mpi4jax/cython/mpi_xla_bridge.pyx (hunk 1 above)
Comment (dionhaefner) [reply mpi4jax/cython/mpi_xla_bridge.pyx:34]: No, `cdef`s must be declared at the start of the function.

**Comment by dionhaefner** [comment]
Done from my side.

**Pull request merged**